inspektor/lint.py

```diff
@@ -83,11 +99,25 @@ def check_dir(self, path):
 
         :param path: Path to a directory.
         """
+        filenames = []
+        failed_paths = []
         for root, dirs, files in os.walk(path):
             for filename in files:
-                self.check_file(os.path.join(root, filename))
-
-        return not self.failed_paths
+                filename = os.path.join(root, filename)
+                checker = PathChecker(path=filename, args=self.args, label='Lint',
+                                      logger=self.log)
+                if checker.check_attributes('text', 'python', 'not_empty'):
+                    filenames.append(os.path.join(root, filename))
+
+        if filenames:
+            pool = multiprocessing.Pool()
+            opts = self.get_opts()
+            args = [(filename, opts) for filename in filenames]
+            failed_paths = pool.map(run_pylint_file, args)
+            failed_paths = [path for path in failed_paths if path is not None]
+            if any(failed_paths):
```

Comment:
I think you combined two approaches here. Either you could always evaluate `failed_paths` and then check `if failed_paths` or you could use `if any(failed_paths)` without processing it (usually it should be `[None, None, None...]` so False)` and only evaluate the `failed_paths` in case of failure. I'd go with the second approach...

Reply:
I mean
```diff
+        if any(failed_paths):
             failed_paths = [path for path in failed_paths if path is not None]
-            if any(failed_paths):
-                self.failed_paths += failed_paths
-        return not failed_paths
+            self.failed_paths += failed_paths
+            return False
+        return True
```